GetAccept-v2/app.js

```diff
@@ -742,39 +779,73 @@
                             var fieldString = fieldData.field_value;
                             var fieldKey = fieldString.replace("{{", "").replace("}}", "");
                             var mergeFieldValues = fieldKey.split('.');
-                            if(mergeFieldValues && mergeFieldValues.length > 1 && mergeFieldValues[0] === className) {
+                            if(mergeFieldValues && mergeFieldValues.length > 1 && viewModel.releatedModels.includes(mergeFieldValues[0])) {
                                 fieldData.field_label = fieldString;
-                                fieldKeyValue = !!mergeFieldValues[1] ? eval('viewModel.'+ className + '["' + mergeFieldValues[1] + '"].text') : '' ;
+                                fieldKeyValue = !!mergeFieldValues[1] ? eval('viewModel.'+ mergeFieldValues[0] + '["' + mergeFieldValues[1] + '"].text') : '' ;
                             }
                             fieldData.field_value = !!fieldKeyValue ? fieldKeyValue : fieldData.field_value;
                         } catch (e) {
                             console.log(e);
                         }
                         viewModel.selectedTemplateFields.push(fieldData);
                     });
-                    viewModel.TemplateFields(true);
+
+                    if(!viewModel.selectedTemplateFields().length && !viewModel.templateRoles().length) {
+                        hideAllSteps();
+                        viewModel.Document(true);
+                    } else {
+                        viewModel.TemplateFields(true);
+                    }
                 } else {
+                    if(viewModel.templateRoles().length){
+                        viewModel.TemplateFields(true);
+                        return;
+                    }
                     viewModel.Document(true);
                 }
             });
         }
 
+        function getTemplateRoles() {
+            viewModel.templateRoles.removeAll();
+            apiRequest("templates/" + viewModel.selectedTemplateId() + '/roles', "GET", "", function (data) {
+                if (data) {
+                    $.each(data.roles, function (index, rolesData) {
+                        try {
+                            viewModel.templateRoles.push(rolesData);
+                        } catch (e) {
```

Comment:
maybe log something more than just error?

Reply:
I think i can remove that